lib/delegate.ml

```diff
@@ -18,11 +18,12 @@ let run_delegate ~dry_run del args =
 
 (* Publish request *)
 
-let publish_distrib ~dry_run ~msg ~archive ~yes pkg =
+let publish_distrib ?token ~dry_run ~msg ~archive ~yes pkg =
```

Comment:
How about moving the token selection further up in `bin` and make this arg mandatory. That makes it clear from the beginning where the token comes from.

Same applies to the opam command, I think the selection should be made as early as possible and then the token passed around rather than delaying the filesystem dependent part.

What do you think?

Reply:
Hmm okay I realize we can't strictly do this because of the delegate case. I guess we'll have to wait until we finally get rid of them!

Reply:
That's what I did at first but yes I realized we shouldn't try to determine the token before calling the delegate as it's specific to github.